### PR TITLE
Show user profile

### PR DESCRIPTION
**Pull request by thiennv57**
Show user profile

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -30,6 +27,21 @@ def update
     end
   end
 
+  def edit
+    @user = User.find(params[:id])
```
Comment (phamkykhoi) [inline app/controllers/users_controller.rb]: Remove ( )

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -30,6 +27,21 @@ def update
...
+  end
+
+  def destroy
+    User.find(params[:id]).destroy
+    flash[:success] = t "user_delete"
+    redirect_to users_url
+  end
+
+  def index
+    @users = User.paginate(page: params[:id], per_page: 10)
```
Comment (phamkykhoi) [inline app/controllers/users_controller.rb]: Remove ( )

Comment (phamkykhoi) [inline app/controllers/users_controller.rb]: Move 10 to Settings

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -30,6 +27,21 @@ def update
...
+
+  end
+
+  def destroy
+    User.find(params[:id]).destroy
+    flash[:success] = t "user_delete"
+    redirect_to users_url
+  end
+
+  def index
```
Comment (phamkykhoi) [inline app/controllers/users_controller.rb]: Move function to head user_controller

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -38,4 +50,22 @@ def set_user
...
+  def logged_in_user
+    unless logged_in?
+      store_location
+      flash[:danger] = t "please_log_in"
+      redirect_to login_url
+    end
+  end
+
+  def correct_user
+    @user = User.find(params[:id])
```
Comment (phamkykhoi) [inline app/controllers/users_controller.rb]: Remove ( )

File: app/views/users/_user.html.erb [hunk 5]
```diff
@@ -0,0 +1,7 @@
+<li>
+  <%= link_to user.name, user %>
+  <% if current_user.admin? && !current_user?(user) %>
+    | <%= link_to "delete", user, method: :delete,
+                                  data: {confirm: "You sure?"} %>
```
Comment (phamkykhoi) [inline app/views/users/_user.html.erb]: I18n

File: app/views/shared/_user_info.html.erb [hunk 6]
```diff
@@ -0,0 +1,2 @@
+<h1><%= current_user.name %></h1>
+<span><%= link_to "view my profile", current_user %></span></span>
```
Comment (phamkykhoi) [inline app/views/shared/_user_info.html.erb]: I18n

File: app/controllers/.fuse_hidden00000beb00000002 [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+      store_location
+      flash[:danger] = t "please_log_in"
+      redirect_to login_url
+    end
+  end
+
+  def correct_user
+    redirect_to(root_url) unless @user == current_user?(@user)
+  end
+end
```
Comment (phamkykhoi) [inline app/controllers/.fuse_hidden00000beb00000002]: Git rm this file

File: app/helpers/sessions_helper.rb [hunk 8]
```diff
@@ -40,4 +40,13 @@ def log_out
     session.delete :user_id
     @current_user = nil
   end
-end
\ No newline at end of file
+
+  def redirect_back_or default 
+    redirect_to(session[:forwarding_url] || default)
+    session.delete(:forwarding_url)
```
Comment (phamkykhoi) [inline app/helpers/sessions_helper.rb]: Remove ( )

File: app/helpers/sessions_helper.rb [hunk 9]
```diff
@@ -40,4 +40,13 @@ def log_out
     session.delete :user_id
     @current_user = nil
   end
-end
\ No newline at end of file
+
+  def redirect_back_or default 
+    redirect_to(session[:forwarding_url] || default)
```
Comment (phamkykhoi) [inline app/helpers/sessions_helper.rb]: Remove ( )

File: app/views/static_pages/home.html.erb [hunk 10]
```diff
@@ -1,31 +1,34 @@
...
+            <%= render "shared/user_info" %>
+          </section>
         </div>
-        <div class="col-md-2">
-          <div class="caption">
-          </div>
+      </div>
+      <div class="col-md-8" style="">
+        <br/>
+        <a href="#" class="action-button shadow animate blue" style="left: 100px;"> <%= t "word_list" %></a>
```
Comment (phamkykhoi) [inline app/views/static_pages/home.html.erb]: Use link_to

File: app/views/static_pages/home.html.erb [hunk 11]
```diff
@@ -1,31 +1,34 @@
...
+        <a href="#" class="action-button shadow animate red" style="left: 200px;"><%= t "lesson" %></a>
```
Comment (phamkykhoi) [inline app/views/static_pages/home.html.erb]: Use link_to

**Comment by dieunb** [comment]
OK

**Pull request merged**